pymess/models/common.py

```diff
@@ -171,15 +170,16 @@ class BaseAbstractTemplate(SmartModel):
     is_active = models.BooleanField(null=False, blank=False, default=True, verbose_name=_('is active'))
     is_allowed_duplicate_messages = models.BooleanField(null=False, blank=False, default=True,
                                                         verbose_name=_('Duplicate messages are allowed'))
-    def _update_context_data(self, context_data):
+
+    def _update_context_data(self, context_data, recipient):
         return context_data
 
-    def render_text_template(self, text, context_data):
-        context_data = self._update_context_data(context_data)
+    def render_text_template(self, text, context_data, recipient):
+        context_data = self._update_context_data(context_data, recipient)
         return Template(text).render(Context(context_data))
 
-    def render_body(self, context_data):
-        return self.render_text_template(self.get_body(), context_data)
+    def render_body(self, context_data, recipient=None):
```

Comment:
I think recipient should be mandatory here, or?

Reply:
In **clean_body** recipient does not exist, so it is optional